src/domain/users/ip-metadata-validator.ts

```diff
@@ -0,0 +1,33 @@
+import { InvalidIpMetadataTypeError } from "@domain/errors"
+import { getBlockListedASNs, getAllowListedCountries } from "@config"
+
+export const IpMetadataValidator = (): IpMetadataValidator => {
+  const validate = (userIP: UserIPs): true | ValidationError => {
+    const blockListedASNs = getBlockListedASNs()
```

Comment:
if we want this to be more unit testable, we should have the blockListedASNs and allowListedCountries passed as a parameter to the IpMetadataValidator wrapper function.